src/backend/mailbox_manager/api/client/viewsets.py

```diff
@@ -392,3 +393,16 @@ def get_queryset(self):
             )
 
         return queryset
+
+    def perform_create(self, serializer):
+        """Lookup for existing user before inviting."""
+        if email := serializer.validated_data["email"]:
+            existing_user = models.User.objects.filter(email=email)
+            if existing_user.exists():
+                return models.MailDomainAccess.objects.create(
+                    user=existing_user[0],
+                    domain=serializer.validated_data["domain"],
+                    role=serializer.validated_data["role"],
+                )
```

Comment:
To make only one query:
```suggestion
            try:
                existing_user = models.User.objects.get(email=email)
            except models.User.DoesNotExist:
                existing_user = None
            if existing_user:
                return models.MailDomainAccess.objects.create(
                    user=existing_user,
                    domain=serializer.validated_data["domain"],
                    role=serializer.validated_data["role"],
                )
```